go/ql/src/Security/CWE-089/StringBreak.qhelp

```diff
@@ -29,17 +31,38 @@ then embeds it into a SQL query built using the Squirrel library.
 </p>
 <sample src="StringBreak.go"/>
 <p>
-Note that while Squirrel provides a structured API for building SQL queries that mitigates against
-common causes of SQL injection vulnerabilities, this code is still vulnerable: if the JSON-encoded
-representation of <code>version</code> contains a single quote, this will prematurely close the
-surrounding string, changing the structure of the SQL expression being constructed. This could be
-exploited to mount a SQL injection attack.
+Note that JSON encoding does not escape single quotes in any way, so this code is vulnerable: any
+single-quote character in <code>version</code> will prematurely close the surrounding string literal,
+changing the structure of the SQL expression being constructed. This could be exploited to mount
+a SQL injection attack.
 </p>
 <p>
-To fix this vulnerability, use Squirrel's placeholder syntax, which avoids the need to explicitly
-construct a quoted string.
+To fix this vulnerability, use the placeholder syntax from Squirrel's structured API for building
+queries, which avoids the need to explicitly construct a quoted string.
 </p>
 <sample src="StringBreakGood.go"/>
+<p>
+In situations where a structured API is not available, make sure to escape quotes before embedding
+user-provided data into a quoted string. For example, this is how you can backslash-escape single
+quotes using <code>strings.ReplaceAll</code>:
+</p>
+<sample language="go">
+  quoted := strings.ReplaceAll(raw, `\`, `\\`)
+  quoted = strings.ReplaceAll(quoted, "'", "\\'")
+</sample>
+<p>
+Note that any existing backslash characters in the string must be escaped first, so that they do
```

Comment:
Would it make sense to update the example above to include both types of escaping, or is this totally obvious to the intended audience?

Reply:
When you say both types of escaping, are you referring to backslash escaping and doubling quotes? I think that's fairly obvious to our audience.